flink-runtime/src/main/java/org/apache/flink/runtime/execution/Environment.java

```diff
@@ -186,7 +186,7 @@ public interface Environment {
     TaskKvStateRegistry getTaskKvStateRegistry();
 
     /**
-     * Confirms that the invokable has successfully completed all steps it needed to to for the
+     * Confirms that the invokable has successfully completed all steps it needed to for the
```

Comment:
Zero `to` might even be better here!
```suggestion
     * Confirms that the invokable has successfully completed all steps it needed for the
```

Reply:
Hum, do you really think so? I'm not a native English speaker, but it seems right to me, judging by reverting the order of the sentence parts: "for the checkpoint, it completed to steps it needed to".